---
title: "CausalPath"
output:
  github_document: default
  html_notebook: default
---
  
```{r knitrSetup, include=FALSE}
library(knitr)
opts_chunk$set(out.extra='style="display:block; margin: auto"', fig.align="center", tidy=FALSE)
verbose <- FALSE
```

# Purpose

Example usage of the CausalPath analysis.

## Publication 

PDF available here: https://www.biorxiv.org/content/early/2018/02/05/258855

## Code and Parameter Description 

Code repository: https://github.com/PathwayAndDataAnalysis/causalpath

# Background 
## Background: Method 
This method generates causal explanations for proteomic profiles using prior mechanistic knowledge in the literature, as recorded in cellular pathway maps. 

# Run CausalPath
## Working directory 

```{r, engine='bash'}
pwd
```

## Parameters 

Parameters are described here: https://github.com/PathwayAndDataAnalysis/causalpath/blob/master/wiki/InputFormat.md

```{r, engine='bash'}
cat dataset/parameters.txt
```

## Run CausalPath 

CausalPath code available from: https://github.com/PathwayAndDataAnalysis/causalpath

```{r, engine='bash'}
java -jar ../java/causalpath-1.2.0_5fa16018.jar dataset
```

# Result Visualization
## Load Libraries 

paxtoolsr Code Repository: https://github.com/BioPAX/paxtoolsr

```{r}
library(paxtoolsr)
library(igraph)
library(rcausalpath)
```

## Causative Network

Loading network generated by CausalPath. NOTE: Columns with extra information (e.g., phosphorylation sites will be discarded during this example). 

```{r}
sif <- readSif("dataset/causative.sif")
head(sif)
```

### Visualize 

NOTE: highlightNodes can be used to "highlight" nodes of interest. 

```{r, fig.width=10, fig.height=10}
g <- loadCausalPathFormat(sif, "dataset/causative.format")
plot(g, vertex.shape="none", vertex.label.cex=1.5, layout=layout.fruchterman.reingold, main="Causative Network")
legend(x="topleft", legend=c("phosphorylation","dephosphorylation","upregulates expression","downregulates expression"), col=c("#006400","#640000","#006400","#640000"), lwd=1, lty=c(2,2,1,1))
```

## Conflicting Network

```{r}
sif <- readSif("dataset/conflicting.sif")
head(sif)
```

### Visualize Network 

```{r, fig.width=10, fig.height=10}
g <- loadCausalPathFormat(sif, "dataset/conflicting.format")
plot(g, vertex.shape="none", vertex.label.cex=1.5, layout=layout.fruchterman.reingold, main="Causative Network")
legend(x="topleft", legend=c("phosphorylation","dephosphorylation","upregulates expression","downregulates expression"), col=c("#006400","#640000","#006400","#640000"), lwd=1, lty=c(2,2,1,1))
```

# Troubleshooting
## Phosphoproteomic Site Locations 

A review of site locations in this work shows that site locations in literature does not always map to the sequence of the canonical protein isoform provided by UniProt. One option is to relax the constraint on exact site matching to allow a difference of 1-2 amino acids. 

# Session Info

```{r}
sessionInfo()
```